### PR TITLE
Add new HostConfig.Labels API to manage custom labels on the static Nodes

### LINKED ISSUE #1822: Allow removal of the ServiceNodeExclusion label  from Master node
Currently this can be manually handle - if this label is removed manually from the node prior to the Nginx ingress controller installation, then only the control plane node is getting added to the load balancer. 

However, to support the KKP setup on single node with minimal manual interventions, it would be good to have this as part of kubeone. 

**What feature would you like to be added?**
Possibility to exclude/remove label from the control plane node using Kubeone

**What are use cases of the feature?**
Use case - In case of single master/worker node, the control plane is not getting added to the backend pool of Nginx Ingress Controller External Load balancer since node contains label  `node.kubernetes.io/exclude-from-external-load-balancers`

### PR DESCRIPTION
**Pull request by kron4eg**
**What type of PR is this?**
/kind feature
/kind api-change

**What this PR does / why we need it**:
To allow better management of the workloads distribution across different Nodes we need ability to add/remove custom labels on the control plane and static worker Nodes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1822

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add new HostConfig.Labels API to manage custom labels on the static Nodes
```

### CODE REVIEW

File: pkg/apis/kubeone/types.go [hunk 1]
```diff
@@ -142,46 +160,63 @@ const (
...
 	// Taints are taints applied to nodes. Those taints are only applied when the node is being provisioned.
 	// If not provided (i.e. nil) for control plane nodes, it defaults to:
 	//   * For Kubernetes 1.23 and older: TaintEffectNoSchedule with key node-role.kubernetes.io/master
 	//   * For Kubernetes 1.24 and newer: TaintEffectNoSchedule with keys
 	//     node-role.kubernetes.io/control-plane and node-role.kubernetes.io/master
 	// Explicitly empty (i.e. []corev1.Taint{}) means no taints will be applied (this is default for worker nodes).
 	Taints []corev1.Taint `json:"taints,omitempty"`
+
+	// Lables to be used to apply (or remove, with minus sign suffix, see more kubectl help label) lables to/from node
+	Labels map[string]string `json:"labels"`
```
Comment (xmudrii) [inline pkg/apis/kubeone/types.go:215]: Why don't we have omitempty here? We have it for other fields.

File: pkg/tasks/nodes.go [hunk 2]
```diff
@@ -131,6 +132,21 @@ func labelNodeOSes(s *state.State) error {
 
 			node.Labels["v1.kubeone.io/operating-system"] = string(host.OperatingSystem)
 
+			for labKey, labVal := range host.Labels {
+				if strings.HasSuffix(labKey, "-") {
+					// additional `if` instead of `if strings.HasSuffix(labKey, "-") && labVal == ""` is used to avoid
+					// entering `else` section when the labVal is not empty.
+					if labVal == "" {
```
Comment (xmudrii) [inline pkg/tasks/nodes.go:139]: I see that if a label has `-` suffix and a value, we ignore it. I think we should have some API validation to reject such cases.

Comment (kron4eg) [reply pkg/tasks/nodes.go]: Good point, fixed

File: pkg/apis/kubeone/types.go (hunk 1 above)
Comment (kron4eg) [reply pkg/apis/kubeone/types.go]: Hm, probably forgotten :shrug:

**Review by xmudrii — CHANGES_REQUESTED** [review]
Can you also please update the example config?

**Review by xmudrii — APPROVED** [review]
/approve

**Pull request merged**